File: 3610-find-x-sum-of-all-k-long-subarrays-i/find-x-sum-of-all-k-long-subarrays-i.py
class Solution:
    def findXSum(self, nums: List[int], k: int, x: int) -> List[int]:
        n = len(nums)
        ans = []

        for i in range(n - k + 1):
            sub = nums[i : i + k]
            freq = Counter(sub)

            most = nlargest(x, freq.items(), key=lambda item: (item[1], item[0]))
            checksum = sum(val * freq for val, freq in most)
            ans.append(checksum)

        return ans

File: 3610-find-x-sum-of-all-k-long-subarrays-i/find-x-sum-of-all-k-long-subarrays-i.cs
public class Solution {
    public int[] FindXSum(int[] nums, int k, int x) {
        int n = nums.Length;
        int[] ans = new int[n - k + 1];

        for (int i = 0; i <= n - k; i++) {
            Dictionary<int, int> cnt = new Dictionary<int, int>();
            for (int j = i; j < i + k; j++) {
                if (cnt.ContainsKey(nums[j])) {
                    cnt[nums[j]]++;
                }
                else {
                    cnt[nums[j]] = 1;
                }
            }

            List<int[]> freq = new List<int[]>();
            foreach (var entry in cnt) {
                freq.Add(new int[] { entry.Value, entry.Key });
            }

            freq.Sort((a, b) => b[0] != a[0] ? b[0] - a[0] : b[1] - a[1]);
            int xsum = 0;

            for (int j = 0; j < x && j < freq.Count; j++) {
                xsum += freq[j][0] * freq[j][1];
            }
            
            ans[i] = xsum;
        }

        return ans;
    }
}